### PR TITLE
Rebase to the latest Ballista version

### PR DESCRIPTION
**Pull request by ch-sc**
Rebasing to the latest Ballista commit to switch to Datafusion 20

- Including 66 commits 
  - Changelog: [0637711...a08347a](https://github.com/coralogix/arrow-ballista/compare/0637711...apache:arrow-ballista:a08347a)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ updates:
     schedule:
       interval: daily
     open-pull-requests-limit: 10
-    target-branch: master
+    target-branch: main
```
Comment (fsdvh) [inline .github/dependabot.yml:25]: we also would need to move our repo to master -> main

File: ballista/executor/src/executor.rs [hunk 2]
```diff
@@ -79,38 +81,21 @@ impl Executor {
...
-            work_dir,
-            runtime,
-            metrics_collector,
-            concurrent_tasks,
-            HashMap::new(),
-            HashMap::new(),
-        )
-    }
-
-    pub fn with_functions(
```
Comment (mpurins-coralogix) [inline ballista/executor/src/executor.rs:94]: We use `Executor::with_functions` and probably want to have it.

File: .github/workflows/comment_bot.yml [hunk 3]
```diff
@@ -58,14 +58,14 @@ jobs:
       - uses: r-lib/actions/pr-fetch@master
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Rebase on ${{ github.repository }} master
+      - name: Rebase on ${{ github.repository }} main
```
Comment (fsdvh) [inline .github/workflows/comment_bot.yml:61]: same

File: ballista/scheduler/src/metrics/prometheus.rs [hunk 4]
```diff
@@ -46,17 +46,14 @@ pub struct PrometheusMetricsCollector {
...
     pending_queue_size: Gauge,
-    active_jobs: Gauge,
 }
 
 impl PrometheusMetricsCollector {
     pub fn new(registry: &Registry) -> Result<Self> {
         let execution_time = register_histogram_with_registry!(
             "job_exec_time_seconds",
             "Histogram of successful job execution time in seconds",
-            vec![
```
Comment (thinkharderdev) [inline ballista/scheduler/src/metrics/prometheus.rs:57]: I think we want to keep this change. We need to setup a way to make this configurable upstream but for now we want different buckets for our specific use case

File: ballista/scheduler/src/metrics/prometheus.rs [hunk 5]
```diff
@@ -66,10 +63,7 @@ impl PrometheusMetricsCollector {
         let planning_time = register_histogram_with_registry!(
             "planning_time_ms",
             "Histogram of job planning time in milliseconds",
-            vec![
```
Comment (thinkharderdev) [inline ballista/scheduler/src/metrics/prometheus.rs:69]: Same here

**Review by thinkharderdev — APPROVED** [review]
One small thing but otherwise LGTM

**Pull request merged**